SmartDeviceLinkTests/SDLSystemCapabilityManagerSpec.m

```diff
@@ -0,0 +1,279 @@
+            testRegisterAppInterfaceResponse.audioPassThruCapabilities = testAudioPassThruCapabilities;
+            testRegisterAppInterfaceResponse.pcmStreamCapabilities = testPCMStreamCapabilities;
+        });
+
+        afterEach(^{
+            SDLRPCResponseNotification *notification = [[SDLRPCResponseNotification alloc] initWithName:SDLDidReceiveRegisterAppInterfaceResponse object:self rpcResponse:testRegisterAppInterfaceResponse];
+
+            [[NSNotificationCenter defaultCenter] postNotification:notification];
+            [NSThread sleepForTimeInterval:0.1];
+        });
+    });
+
+    context(@"When notified of a set display layout response", ^ {
+       __block SDLSetDisplayLayoutResponse *testSetDisplayLayoutResponse = nil;
+
+        it(@"should not save any capabilities if all the capabilities in the set display layout response are nil", ^{
+            testSetDisplayLayoutResponse = [[SDLSetDisplayLayoutResponse alloc] init];
+            testSetDisplayLayoutResponse.displayCapabilities = nil;
+            testSetDisplayLayoutResponse.buttonCapabilities = nil;
+            testSetDisplayLayoutResponse.softButtonCapabilities = nil;
+            testSetDisplayLayoutResponse.presetBankCapabilities = nil;
+        });
+
+        it(@"should should save the correct capabilities if all capabilities in the set display layout response are valid", ^{
+            testDisplayCapabilities = [[SDLDisplayCapabilities alloc] init];
+            testDisplayCapabilities.graphicSupported = @NO;
+
+            SDLSoftButtonCapabilities *softButtonCapability = [[SDLSoftButtonCapabilities alloc] init];
+            softButtonCapability.shortPressAvailable = @NO;
+            softButtonCapability.longPressAvailable = @NO;
+            softButtonCapability.upDownAvailable = @NO;
+            softButtonCapability.imageSupported = @NO;
+            testSoftButtonCapabilities = @[softButtonCapability];
+
+            SDLButtonCapabilities *buttonCapabilities = [[SDLButtonCapabilities alloc] init];
+            buttonCapabilities.name = SDLButtonNameOk;
+            buttonCapabilities.shortPressAvailable = @NO;
+            buttonCapabilities.longPressAvailable = @NO;
+            buttonCapabilities.upDownAvailable = @NO;
+            testButtonCapabilities = @[buttonCapabilities];
+
+            testPresetBankCapabilities = [[SDLPresetBankCapabilities alloc] init];
+            testPresetBankCapabilities.onScreenPresetsAvailable = @NO;
+
+            testSetDisplayLayoutResponse = [[SDLSetDisplayLayoutResponse alloc] init];
+            testSetDisplayLayoutResponse.displayCapabilities = testDisplayCapabilities;
+            testSetDisplayLayoutResponse.buttonCapabilities = testButtonCapabilities;
+            testSetDisplayLayoutResponse.softButtonCapabilities = testSoftButtonCapabilities;
+            testSetDisplayLayoutResponse.presetBankCapabilities = testPresetBankCapabilities;
+        });
+
+        afterEach(^{
```

Comment:
I guess what I'm wondering is why you're using `afterEach` so much instead of `beforeEach`

Reply:
Don't `beforeEach` and `afterEach` both exist to avoid writing redundant code? Each test case only has two `afterEach`s that get's called which doesn't seem like a lot to me.

Reply:
No, my question is what the purpose of `afterEach` is rather than using a `beforeEach` in this case.